packages/context-menu/src/vaadin-context-menu.js

```diff
@@ -443,22 +438,19 @@ class ContextMenu extends ControllerMixin(ElementMixin(ThemePropertyMixin(ItemsM
 
   /** @private */
   _closeOnChanged(closeOn, oldCloseOn) {
-    // Listen on this.$.overlay.root to workaround issue on
-    //  ShadyDOM polyfill: https://github.com/webcomponents/shadydom/issues/159
-
     // Outside click event from overlay
     const evtOverlay = 'vaadin-overlay-outside-click';
 
+    const overlay = this.$.overlay;
+
     if (oldCloseOn) {
-      this._unlisten(this.$.overlay, oldCloseOn, this._boundClose);
-      this._unlisten(this.$.overlay.root, oldCloseOn, this._boundClose);
+      this._unlisten(overlay, oldCloseOn, this._boundClose);
     }
     if (closeOn) {
-      this._listen(this.$.overlay, closeOn, this._boundClose);
-      this._listen(this.$.overlay.root, closeOn, this._boundClose);
-      this._unlisten(this.$.overlay, evtOverlay, this._preventDefault);
+      this._listen(overlay, closeOn, this._boundClose);
+      overlay.removeEventListener(evtOverlay, this._boundPreventDefault);
```

Comment:
No need to use `_listen` as `'vaadin-overlay-outside-click` is not a gesture event.